### PR TITLE
Add a compacter message view mode; closes #2906

### LINKED ISSUE #2906: Make the "compact mode" really compact
Your Rocket.Chat version: 0.26

The "compact mode" must be more compact to fit the small netbook's screens (1366*768). So it would be nice to hide avatars, show smaller previews (80px height), make the interline space smaller and so on:

image

### PR DESCRIPTION
**Pull request by lucasvanhalst**
@RocketChat/core 

Closes  #2906

Adds 3 different message view modes (instead of only normal/compact):

Most notable is the new 'compact' option, which makes the messages really compact, IRC-style.
The old 'compact' mode is still available, but renamed to 'cozy'

Before: 
After: 

This does hide the admin/mod etc tags though to save some space, but maybe we should add an option to show/hide it on the admin pages

Also adds an additional option in the user preferences to show or hide usernames in the message template, to potentially save some extra space (could be useful on mobile devices for example).

### CODE REVIEW

File: server/startup/migrations/v044.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+RocketChat.Migrations.add({
+	version: 44,
+	up: function() {
+		// Set default message viewMode to 'normal' or 'cozy' depending on the users' current settings and remove the field 'compactView'
+		var normalViewUserIds = [],
+			cozyViewUserIds = [];
+		RocketChat.models.Users.find().forEach(
+			function(user){
+				if(user.settings && user.settings.preferences && user.settings.preferences.compactView){
+					cozyViewUserIds.push(user._id)
```
Comment (rodrigok) [inline server/startup/migrations/v044.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2416760355/issues/source?bid=3199634&fileBranchId=3237220#l10)

File: server/startup/migrations/v044.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+				}else{
+					normalViewUserIds.push(user._id)
```
Comment (rodrigok) [inline server/startup/migrations/v044.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2416760355/issues/source?bid=3199634&fileBranchId=3237220#l12)

**Comment by rodrigok** [comment]
What about use only avatars?

68747470733a2f2f692e696d6775722e636f6d2f676645673333772e706e67 2

**Comment by lvh1** [comment]
Hmm I don't know, could be kinda confusing since people tend to not upload an avatar (at least on my server)

**Comment by engelgabriel** [comment]
Only if you can hoover the avatar and see the username, but this should be configurable and not the default.

**Comment by lvh1** [comment]
I've cleaned up the commit history a bit, and added the option you suggested @engelgabriel

File: server/startup/migrations/v046.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	version: 46,
+	up: function() {
+		// Set default message viewMode to 'normal' or 'cozy' depending on the users' current settings and remove the field 'compactView'
+		var normalViewUserIds = [],
+			cozyViewUserIds = [];
+		RocketChat.models.Users.find().forEach(
+			function(user){
+				if(user.settings && user.settings.preferences && user.settings.preferences.compactView){
+					cozyViewUserIds.push(user._id);
+				}else{
```
Comment (rodrigok) [inline server/startup/migrations/v046.js]: Codacy Issue found: [Missing space before opening brace.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2454581114/issues/source?bid=3199634&fileBranchId=3237220#l11)

Comment (rodrigok) [inline server/startup/migrations/v046.js]: Codacy Issue found: [Missing space before keyword "else".](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2454581114/issues/source?bid=3199634&fileBranchId=3237220#l11)

File: server/startup/migrations/v046.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+RocketChat.Migrations.add({
+	version: 46,
+	up: function() {
+		// Set default message viewMode to 'normal' or 'cozy' depending on the users' current settings and remove the field 'compactView'
+		var normalViewUserIds = [],
+			cozyViewUserIds = [];
+		RocketChat.models.Users.find().forEach(
+			function(user){
```
Comment (rodrigok) [inline server/startup/migrations/v046.js]: Codacy Issue found: [Missing space before opening brace.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2454581114/issues/source?bid=3199634&fileBranchId=3237220#l8)

File: server/startup/migrations/v046.js (hunk 3 above)
Comment (rodrigok) [inline server/startup/migrations/v046.js]: Codacy Issue found: [Keyword "else" must be followed by whitespace.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2454581114/issues/source?bid=3199634&fileBranchId=3237220#l11)

File: server/startup/migrations/v046.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+				if(user.settings && user.settings.preferences && user.settings.preferences.compactView){
```
Comment (rodrigok) [inline server/startup/migrations/v046.js]: Codacy Issue found: [Missing space before opening brace.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2454581114/issues/source?bid=3199634&fileBranchId=3237220#l9)

Comment (rodrigok) [inline server/startup/migrations/v046.js]: Codacy Issue found: [Keyword "if" must be followed by whitespace.](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2454581114/issues/source?bid=3199634&fileBranchId=3237220#l9)

**Comment by marceloschmidt** [comment]
@lucasvanhalst did you add the hover to show the username?

**Comment by lucasvanhalst** [comment]
@marceloschmidt I think so https://github.com/RocketChat/Rocket.Chat/pull/2913/files#diff-da06ed7f6011dd615644ab7124c33ff9R250

**Comment by marceloschmidt** [comment]
Thanks! I'll have a look at why it's not working on my instance ;)

**Pull request merged**